src/Microsoft.AspNetCore.Server.Kestrel/KestrelServer.cs

```diff
@@ -164,9 +164,14 @@ public void Start<TContext>(IHttpApplication<TContext> application)
                     {
                         var parsedAddress = ServerAddress.FromUrl(address);
 
+                        if (parsedAddress.Scheme.Equals("https", StringComparison.OrdinalIgnoreCase))
+                        {
+                            throw new InvalidOperationException($"HTTPS addresses are not supported. Use {nameof(KestrelServerOptions)}.{nameof(KestrelServerOptions.Listen)}() to configure an HTTPS endpoint.");
+                        }
+
                         if (!string.IsNullOrEmpty(parsedAddress.PathBase))
```

Comment:
There's a `ServerAddress` test that checks that just '/' is not set as path base: https://github.com/aspnet/KestrelHttpServer/blob/632780dd16729298933f96796e6a969f3470d810/test/Microsoft.AspNetCore.Server.KestrelTests/ServerAddressTests.cs#L42